public/app/ml-worker-fetch-cache.js

```
@@ -533,6 +572,21 @@ function _finalizeProgressPanel(results) {
 window.addEventListener('vip:modelDownloadProgress', (e) => {
   const { key, pct } = e.detail;
   _updateProgressRow(key, pct, pct >= 0 ? `${pct}%` : 'Streaming...');
+
+  // If the panel was dismissed (post-startup on-demand download), re-surface
+  // progress in the processing overlay so the user isn't staring at a frozen
+  // screen while a large model (e.g. demucs ~87 MB) downloads mid-pipeline.
+  if (!document.getElementById('vip-model-load-panel') && window.VIPOverlay && typeof window.VIPOverlay.update === 'function') {
+    const displayName = MODEL_DISPLAY_NAMES[key] || key;
+    const pctLabel    = pct >= 0 ? `${pct}%` : '…';
+    try {
+      window.VIPOverlay.update(
```

ESLint

[error] 'window' is not defined.

(no-undef)